crates/wadm-types/src/validation.rs

```diff
@@ -678,6 +679,43 @@ pub fn validate_component_properties(application: &Manifest) -> Vec<ValidationFa
     failures
 }
 
+/// Funtion to validate the component configs
+/// from 0.13.0 source_config is deprecated and replaced with source:config:
+/// this function validates the component configs
+pub fn validate_component_configs(application: &Manifest) -> Vec<ValidationFailure> {
+    let mut failures = Vec::new();
+    for component in application.spec.components.iter() {
+        if has_deprecated_trait_property(component) {
+            failures.push(ValidationFailure::new(
+                ValidationFailureLevel::Error,
```

Comment:
This should be warning level as the field is just deprecated. This would be a change in the test as well to check for warnings